### ISSUE TITLE
[SDF] Tracking Issue

### ISSUE THREAD

**Issue by mole99** [steveicarus/iverilog#943]
This issue tracks all open issues regarding Standard Delay Format (SDF) support in Icarus Verilog. Additionally, issues for `specify` are listed as they are often related.

**SDF**

- [ ] #746
- [ ] #509
- [ ] #899

**Specify**

- [ ] #107

#### Source Forge Issues:

**SDF**

- SDF back annotation fails/aborts for a tri-state buffer https://sourceforge.net/p/iverilog/bugs/960/
- No support for annotating specparams from SDF file https://sourceforge.net/p/iverilog/bugs/889/

**Specify**

- specify block delays break when multiple inputs have the same driver https://sourceforge.net/p/iverilog/bugs/954/
- Unexpected race of modpath "if" expressions (pr1877743.v) https://sourceforge.net/p/iverilog/bugs/495/
- Path delay expression without parenthesis not supported https://sourceforge.net/p/iverilog/bugs/493/
- Module path delays do not properly filter pulses https://sourceforge.net/p/iverilog/bugs/888/

**Comment by caryr** [comment]
For the pieces that are not implemented or working correctly you may want to consider parsing them and ignoring them after emitting a warning either in the RTL parser or the SDF parser. You can then get the compiler to handle the RTL part correctly and have the targets emit the warning or do the correct thing.

There is also the implementation of path delays which is needed for SDF interconnect delays. I think there is an old PR for that.

**Comment by mole99** [comment]
Yes, that's what I would like to do for every unsupported feature. If you know any unsupported feature that needs a warning just tell me and I will add it.

Do you maybe recall what this PR was named? I searched with the keywords `path`, `delay` and `interconnect` but could not find anything.

**Comment by caryr** [comment]
Here are some other SF PRs:

https://sourceforge.net/p/iverilog/bugs/740/ - timing checks
https://sourceforge.net/p/iverilog/bugs/273/ - wire delays

There are also a couple other PRs that will be fixed when SF-495 is fixed. I can help you track those down later.

**Comment by mole99** [comment]
Thanks for the issues, I have added them to the list.

It seems we have a different understanding of what a PR is :)
Pull Request (PR) is GitHub jargon for when you want to add new features to a branch.
If you are using GitLab, it is called Merge Request (MR) there.

See here for a better explanation: [pull-request-vs-merge-request](https://stackoverflow.com/questions/22199432/pull-request-vs-merge-request)

**Comment by caryr** [comment]
If you notice many of the tests start or have pr in their title. I believe this is done to indicate the problem report they where initially ties to. We added gh to indicate they are from github instead of SourceForge. It's sometimes useful when we need to track a change back to an initial request. Of course it didn't help when SF reordered the reports to be project specific.

**Comment by mole99** [comment]
Oh, I see! I have wondered why they are named that way ^^

**Comment by martinwhitaker** [comment]
Would it be better to add a new "SDF" label rather than have a tracking issue? You could then use the filter expression "label:SDF" to find all the relevant issues.

**Comment by mole99** [comment]
That's a great idea! But I don't think I have the permission to create and assign labels. Maybe someone else could do that?

**Comment by caryr** [comment]
I added a specify/SDF tag. I marked the issues on the first page using this tag. We need to mark the rest and then this can be closed except this report does still have some benefit since it references the SourceForge bug reports.
